edn/src/types.rs

```diff
@@ -398,6 +398,12 @@ macro_rules! def_common_value_impl {
             def_common_value_methods!($t<$tchild>);
         }
 
+        impl AsRef<$t> for $t {
```

Comment:
This really doesn't make much sense; it's equivalent to `&x`.  You say it's convenient, but can you explain why?